src/chrome/chromeDebugAdapter.ts

```diff
@@ -449,32 +453,39 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
      * This event tells the client to begin sending setBP requests, etc. Some consumers need to override this
      * to send it at a later time of their choosing.
      */
-    protected sendInitializedEvent(): void {
+    protected async sendInitializedEvent(): Promise<void> {
         // Wait to finish loading sourcemaps from the initial scriptParsed events
         if (this._initialSourceMapsP) {
             const initialSourceMapsP = this._initialSourceMapsP;
             this._initialSourceMapsP = null;
 
-            initialSourceMapsP.then(() => {
+            await initialSourceMapsP.then(async () => {
                 this._session.sendEvent(new InitializedEvent());
-                this._earlyScripts.forEach(script => this.sendLoadedSourceEvent(script));
+                await Promise.all(this._earlyScripts.map(async script => await this.sendLoadedSourceEvent(script)));
                 this._earlyScripts = null;
             });
         }
     }
 
+    public doAfterProcessingSourceEvents(action: () => void): Promise<void> {
```

Comment:
`action` returns a Promise

Reply:
The use inside onExecutionContextsCleared() doesn't return a promise